File: solutions/python/Jewels and Stones.py
class Solution:
    def getCount(self,jewel:str,stones:str)->int:
            [i,j,count]=[0,len(stones)-1,0]
            while  i<=j:
                if i==j:
                    if stones[i]==jewel:
                        count=count+1
                else:
                    if stones[i]==jewel:
                        count=count+1
                    if stones[j]==jewel:
                        count=count+1
                i=i+1
                j=j-1
            return count

    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        [i,j,count]=[0,len(jewels)-1,0]
        while i<=j:
            if i==j:
                count=count+self.getCount(jewels[i],stones)
            else:
                count=count+self.getCount(jewels[i],stones)
                count=count+self.getCount(jewels[j],stones)
            i=i+1
            j=j-1
        return count

File: solutions/cpp/Jewels and Stones.cpp
#include <string>
class Solution
{
public:
    int getCount(char jewel, std::string stones)
    {
        int i = 0, j = stones.length() - 1, count = 0;
        while (i <= j)
        {
            if (i == j)
            {
                if (stones[i] == jewel)
                    count = count + 1;
            }
            else
            {
                if (jewel == stones[i])
                    count = count + 1;
                if (jewel == stones[j])
                    count = count + 1;
            }
            i = i + 1;
            j = j - 1;
        }
        return count;
    }
    int numJewelsInStones(std::string jewels, std::string stones)
    {
        int i = 0, j = jewels.length() - 1, count = 0;
        while (i <= j)
        {
            if (i == j)
                count = count + getCount(jewels[i], stones);
            else
            {
                count = count + getCount(jewels[i], stones);
                count = count + getCount(jewels[j], stones);
            }
            i = i + 1;
            j = j - 1;
        }
        return count;
    }
};